src/docset.ts

```diff
@@ -26,6 +29,119 @@ export type DocSetData<T> = {
 export type SignedDocSet = DocSetData<Array<DagJWSResult>>
 export type EncodedSignedDocSet = DocSetData<Array<EncodedDagJWSResult>>
 
+// TODO: import these types from IDX GraphQL lib once published
+type FieldCommon = { required?: boolean }
+type FieldType =
+  | { type: 'boolean' }
+  | { type: 'integer' }
+  | { type: 'float' }
+  | { type: 'string' }
+  | { type: 'list'; name: string }
+  | { type: 'object'; name: string }
+  | { type: 'reference'; name: string }
+
+type Field = FieldCommon & FieldType
+
+type ObjectFields = Record<string, Field>
+
+type DocReference = {
+  id: string
+  schema: string
+}
+
+type GraphQLDocSetRecords = {
+  index: Record<string, DocReference>
+  lists: Record<string, string>
+  nodes: Record<string, string>
+  objects: Record<string, ObjectFields>
+  references: Record<string, Array<string>>
+  roots: Record<string, DocReference>
+}
+
+type AddSchemaOptions = {
+  name?: string
+  prefix: string
+}
+
+function addSchema(
```

Comment:
Can you describe what this and the other new function in this class does?

Reply:
Sure, as comments in the code or in this PR?

Reply:
This PR, I just want to understand what changes you introduced.

Reply:
Sure, basically this is mostly preparatory work for docsets, having a single instance that can handle schemas, definitions and individual tiles.
The idea is to cover the use case we have for the IDX schemas and definitions than need to be published but in a generic way, and with a simpler DX (for example adding a definition to a docset will also add its schema).
There are also some functions to import/export the signed records from/to a Ceramic node.
It's mostly work in progress and lacking tests but it works for simple cases so far.